Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -2129,6 +2292,103 @@ static void appendPipelineAppSinkCapsChanged(GObject*, GParamSpec*, AppendPipeli
+        gsize bufferSize = gst_buffer_get_size(buffer);
+
+        TRACE_MEDIA_MESSAGE("buffer of size %" G_GSIZE_FORMAT " going thru", bufferSize);
+
+        if (bufferSize > 0)
+            appendPipeline->reportEndOfAppendDataMarkNeeded();
+
+        return GST_PAD_PROBE_OK;
+    }
+
+#ifdef DEBUG_APPEND_PIPELINE_PADS
+    if (GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM) {
+        GstEvent* event = GST_PAD_PROBE_INFO_EVENT(info);
+        if (GST_EVENT_TYPE(event) != GST_EVENT_CUSTOM_DOWNSTREAM)
+            return GST_PAD_PROBE_OK;
+
+        const GstStructure* structure = gst_event_get_structure(event);
+        if (!gst_structure_has_name(structure, "end-of-append-data-mark"))
+            return GST_PAD_PROBE_OK;
+
+        guint id = gst_event_get_seqnum(event);
+        TRACE_MEDIA_MESSAGE("custom downstream event id=%u", id);
+
+        return GST_PAD_PROBE_OK;
+    }
+#endif
+
+    ASSERT_NOT_REACHED();
+    return GST_PAD_PROBE_OK;
+}
+
+static GstPadProbeReturn appendPipelineAppsinkDataEntering(GstPad*, GstPadProbeInfo* info, AppendPipeline* appendPipeline)
+{
+    if (GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM) {
+        GstEvent* event = GST_PAD_PROBE_INFO_EVENT(info);
+        if (GST_EVENT_TYPE(event) != GST_EVENT_CUSTOM_DOWNSTREAM)
+            return GST_PAD_PROBE_OK;
+
+        const GstStructure* structure = gst_event_get_structure(event);
+        if (!gst_structure_has_name(structure, "end-of-append-data-mark"))
+            return GST_PAD_PROBE_OK;
+
+        guint id = gst_event_get_seqnum(event);
+
+        TRACE_MEDIA_MESSAGE("id=%u", id);
+
+        appendPipeline->reportEndOfAppendDataMarkReceived(id);
+
+        return GST_PAD_PROBE_OK;
+    }
+
+#ifdef DEBUG_APPEND_PIPELINE_PADS
+    if (GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_BUFFER) {
+        GstBuffer* buffer = GST_PAD_PROBE_INFO_BUFFER(info);
+        TRACE_MEDIA_MESSAGE("buffer of size %" G_GSIZE_FORMAT " going thru", gst_buffer_get_size(buffer));
+        return GST_PAD_PROBE_OK;
+    }
+#endif
+
+    ASSERT_NOT_REACHED();
+    return GST_PAD_PROBE_OK;
+}
+
+#ifdef DEBUG_APPEND_PIPELINE_PADS
+static GstPadProbeReturn appendPipelinePadProbeDebugInformation(GstPad*, GstPadProbeInfo* info, struct PadProbeInformation* padProbeInformation)
+{
+    ASSERT(GST_PAD_PROBE_INFO_TYPE(info) != static_cast<GstPadProbeType>(GST_PAD_PROBE_TYPE_BUFFER | GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM));
+    if (GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_BUFFER) {
+        GstBuffer* buffer = GST_PAD_PROBE_INFO_BUFFER(info);
+        TRACE_MEDIA_MESSAGE("%s: buffer of size %" G_GSIZE_FORMAT " going thru", padProbeInformation->m_description, gst_buffer_get_size(buffer));
+        return GST_PAD_PROBE_OK;
+    }
+
+    if (GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM) {
+        GstEvent* event = GST_PAD_PROBE_INFO_EVENT(info);
+        if (GST_EVENT_TYPE(event) != GST_EVENT_CUSTOM_DOWNSTREAM)
```

Comment:
There are different types of custom downstream events, they can be CUSTOM_DOWNSTREAM, CUSTOM_DOWNSTREAM_OOB and CUSTOM_DOWNSTREAM_STICKY and this line affects to the event type. The one two lines above affects to the probe type which considers the three event types as one.

Anyway, it is true that most likely we are getting only our own CUSTOM_DOWNSTREAM events and this could be reworked accordingly (removing or turning it into an ASSERT).